flyway: Broken since 9.18.0

### Describe the bug
The application `flyway` has been broken since version 9.21.0.
The previous and last working version was 9.18.0 (d457cf2601d73dbfb047e874221d76cbb74f43ab).

### Steps To Reproduce
Steps to reproduce the behavior:

Simply call `flyway --version`, and you will be greeted with the error message: `java.lang.NoClassDefFoundError`.

For the current version (9.21.2) run:

`nix-shell -p flyway -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/master.tar.gz --command "flyway --version"`

Full output:

```
Exception in thread "main" java.lang.NoClassDefFoundError: com/fasterxml/jackson/core/Versioned
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1013)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:862)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:760)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:681)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:639)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at org.flywaydb.commandline.Main.main(Main.java:107)
Caused by: java.lang.ClassNotFoundException: com.fasterxml.jackson.core.Versioned
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 10 more
```


### Expected behavior
To get the correct version output of `flyway`

### Notify maintainers

@cmcdragonkai

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.4.12, NixOS, 23.11 (Tapir), 23.11.20230827.cddebdb`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.0`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

--- a/pkgs/development/tools/flyway/default.nix
+++ b/pkgs/development/tools/flyway/default.nix
@@ -1,9 +1,10 @@
-{ lib, stdenv, fetchurl, jre_headless, makeWrapper }:
-stdenv.mkDerivation rec{
+{ lib, stdenv, fetchurl, jre_headless, makeWrapper, testers }:
+
+stdenv.mkDerivation (finalAttrs: {
   pname = "flyway";
   version = "9.21.2";
   src = fetchurl {
-    url = "mirror://maven/org/flywaydb/flyway-commandline/${version}/flyway-commandline-${version}.tar.gz";
+    url = "mirror://maven/org/flywaydb/flyway-commandline/${finalAttrs.version}/flyway-commandline-${finalAttrs.version}.tar.gz";
     sha256 = "sha256-7MIjXF1qgEye2Z/cyeuSFkEmXT8AxkwYfw+/UevsPFg=";
   };
   nativeBuildInputs = [ makeWrapper ];
@@ -19,6 +20,9 @@ stdenv.mkDerivation rec{
       --add-flags "org.flywaydb.commandline.Main" \
       --add-flags "-jarDirs='$out/share/flyway/jars'"
   '';
+  passthru.tests = {
+    version = testers.testVersion { package = finalAttrs.finalPackage; };
+  };
   meta = with lib; {
     description = "Evolve your Database Schema easily and reliably across all your instances";
     longDescription = ''
@@ -35,4 +39,4 @@ stdenv.mkDerivation rec{
     platforms = platforms.unix;
     maintainers = [ maintainers.cmcdragonkai ];
   };
-}
+})

--- a/pkgs/development/tools/flyway/default.nix
+++ b/pkgs/development/tools/flyway/default.nix
@@ -2,10 +2,10 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "flyway";
-  version = "9.21.2";
+  version = "9.22.0";
   src = fetchurl {
     url = "mirror://maven/org/flywaydb/flyway-commandline/${finalAttrs.version}/flyway-commandline-${finalAttrs.version}.tar.gz";
-    sha256 = "sha256-7MIjXF1qgEye2Z/cyeuSFkEmXT8AxkwYfw+/UevsPFg=";
+    sha256 = "sha256-vdg66ETDfa0WG1TrRWJ9XpMSCE9sd5GlYAJY2TERC5Y=";
   };
   nativeBuildInputs = [ makeWrapper ];
   dontBuild = true;

--- a/pkgs/development/tools/flyway/default.nix
+++ b/pkgs/development/tools/flyway/default.nix
@@ -12,7 +12,7 @@ stdenv.mkDerivation rec{
   installPhase = ''
     mkdir -p $out/bin $out/share/flyway
     cp -r sql jars drivers conf $out/share/flyway
-    install -Dt $out/share/flyway/lib lib/community/*.jar lib/*.jar
+    install -Dt $out/share/flyway/lib lib/community/*.jar lib/*.jar lib/aad/*.jar lib/oracle_wallet/*.jar
     makeWrapper "${jre_headless}/bin/java" $out/bin/flyway \
       --add-flags "-Djava.security.egd=file:/dev/../dev/urandom" \
       --add-flags "-classpath '$out/share/flyway/lib/*:$out/share/flyway/drivers/*'" \

--- a/pkgs/development/tools/flyway/default.nix
+++ b/pkgs/development/tools/flyway/default.nix
@@ -34,6 +34,7 @@ stdenv.mkDerivation (finalAttrs: {
     '';
     downloadPage = "https://github.com/flyway/flyway";
     homepage = "https://flywaydb.org/";
+    changelog = "https://documentation.red-gate.com/fd/release-notes-for-flyway-engine-179732572.html";
     sourceProvenance = with sourceTypes; [ binaryBytecode ];
     license = licenses.asl20;
     platforms = platforms.unix;

